src/main/kotlin/com/team4099/robot2021/subsystems/Climber.kt

```diff
@@ -0,0 +1,52 @@
+package com.team4099.robot2021.subsystems
+
+import com.revrobotics.CANSparkMax
+import com.revrobotics.CANSparkMaxLowLevel
+import com.revrobotics.ControlType
+import com.team4099.lib.units.base.meters
+import com.team4099.lib.units.perSecond
+import com.team4099.lib.units.sparkMaxLinearMechanismSensor
+import com.team4099.robot2021.config.Constants
+import edu.wpi.first.wpilibj.Solenoid
+import edu.wpi.first.wpilibj2.command.SubsystemBase
+
+object Climber: SubsystemBase() {
+  private val climberRArm = CANSparkMax(Constants.Climber.CLIMBERRARM_SPARKMAX_ID, CANSparkMaxLowLevel.MotorType.kBrushless)
+  private val climberLArm = CANSparkMax(Constants.Climber.CLIMBERlARM_SPARKMAX_ID, CANSparkMaxLowLevel.MotorType.kBrushless)
+  private val climberRArmPIDController = climberRArm.pidController
+  private val climberLArmPIDController = climberLArm.pidController
+  private val climberRArmSensor = sparkMaxLinearMechanismSensor(climberRArm, Constants.Climber.CLIMBER_SENSOR_LINEARMECH_GEARRATIO, Constants.Climber.CLIMBER_SENSOR_LINEARMECH_PULLEYDIAMETER)//diameter: .0508 meters = 2 in
+  private val climberLArmSensor = sparkMaxLinearMechanismSensor(climberLArm, Constants.Climber.CLIMBER_SENSOR_LINEARMECH_GEARRATIO, Constants.Climber.CLIMBER_SENSOR_LINEARMECH_PULLEYDIAMETER)//diameter: .0508 meters = 2 in
+  private val pneumaticRBrake = Solenoid(Constants.Climber.CLIMBER_SOLENOID_ACTUATIONSTATE) //unactuated state is having the pneumatic extended out to lock
+  private val pneumaticLBrake = Solenoid(Constants.Climber.CLIMBER_SOLENOID_ACTUATIONSTATE) //unactuated state is having the pneumatic extended out to lock
+  var brakeApplied = false
+    set(value) {
+      field = value
+      pneumaticRBrake.set(value)
+      pneumaticLBrake.set(value)
+    }
+
+  init {
+    climberRArmPIDController.p = Constants.Climber.CLIMBER_CLIMBERPIDCONTROLLER_P
+    climberRArmPIDController.i = Constants.Climber.CLIMBER_CLIMBERPIDCONTROLLER_I
+    climberRArmPIDController.d = Constants.Climber.CLIMBER_CLIMBERPIDCONTROLLER_D
+    climberRArmPIDController.setSmartMotionMaxVelocity(climberRArmSensor.velocityToRawUnits(0.5.meters.perSecond), Constants.Climber.CLIMBER_PID_SLOTID_SMARTMOTIIONVEL) //what is the slotID (second argument)
```

Comment:
make the velocity/acceleration you pass here in constants

Reply:
also the slot id can be hardcoded to 0 for all of this

there's different slots on the motor controller for PID gains because sometimes you need multiple sets, but we never specify a slot anywhere else so it defaults to 0, and there's no reason to move it to constants where things could possibly be screwed up